plugin-name/plugin-name.php

```diff
@@ -30,46 +30,45 @@
 	die;
 }
 
-/**
- * The code that runs during plugin activation.
- * This action is documented in includes/class-plugin-name-activator.php
- */
-function activate_plugin_name() {
-	require_once plugin_dir_path( __FILE__ ) . 'includes/class-plugin-name-activator.php';
-	Plugin_Name_Activator::activate();
-}
+class Plugin_Name_Init {
 
-/**
- * The code that runs during plugin deactivation.
- * This action is documented in includes/class-plugin-name-deactivator.php
- */
-function deactivate_plugin_name() {
-	require_once plugin_dir_path( __FILE__ ) . 'includes/class-plugin-name-deactivator.php';
-	Plugin_Name_Deactivator::deactivate();
-}
+	/**
+	 * The code that runs during plugin activation.
+	 * This action is documented in includes/class-plugin-name-activator.php
+	 */
+	function activate_plugin_name() {
+		require_once plugin_dir_path( __FILE__ ) . 'includes/class-plugin-name-activator.php';
+		Plugin_Name_Activator::activate();
+	}
 
-register_activation_hook( __FILE__, 'activate_plugin_name' );
-register_deactivation_hook( __FILE__, 'deactivate_plugin_name' );
+	/**
+	 * The code that runs during plugin deactivation.
+	 * This action is documented in includes/class-plugin-name-deactivator.php
+	 */
+	function deactivate_plugin_name() {
+		require_once plugin_dir_path( __FILE__ ) . 'includes/class-plugin-name-deactivator.php';
+		Plugin_Name_Deactivator::deactivate();
+	}
 
-/**
- * The core plugin class that is used to define internationalization,
- * admin-specific hooks, and public-facing site hooks.
- */
-require plugin_dir_path( __FILE__ ) . 'includes/class-plugin-name.php';
+	/**
+	 * Bootstrap.
+	 */
+	public static function init() {
 
-/**
- * Begins execution of the plugin.
- *
- * Since everything within the plugin is registered via hooks,
- * then kicking off the plugin from this point in the file does
- * not affect the page life cycle.
- *
- * @since    1.0.0
- */
-function run_plugin_name() {
+		register_activation_hook( __FILE__, array('Plugin_Name_Init', 'activate_quiz_reports') );
+		register_deactivation_hook( __FILE__, array('Plugin_Name_Init', 'deactivate_quiz_reports') );
 
-	$plugin = new Plugin_Name();
-	$plugin->run();
+		/**
+		 * The core plugin class that is used to define internationalization,
+		 * admin-specific hooks, and public-facing site hooks.
+		 */
+		require plugin_dir_path( __FILE__ ) . 'includes/class-plugin-name.php';
+
+		$plugin = new Plugin_Name();
+		$plugin->run();
+
+	}
 
 }
-run_plugin_name();
+
+add_action( 'init', array( 'Plugin_Name_Init', 'init' ) );
```

Comment:
I believe this would mean that the plugin won't load until the `init` hook, which would be a problem if the plugin is using any hooks earlier than `init`.